contrib/mesos/docs/discovery.md

```diff
@@ -0,0 +1,88 @@
+# Discovery
+
+## DNS
+
+### kube-dns
+
+[**kube-dns**](https://github.com/kubernetes/kubernetes/blob/release-1.1/docs/admin/dns.md) is a Kubernetes add-on that works out of the box with Kubernetes-Mesos.
+For details on usage see the implementation in the `cluster/mesos/docker` source tree.
+kube-dns provides records both for services and pods.
+
+### mesos-dns
+
+**NOTE:** There is still no support for publishing Kubernetes *services* in mesos-dns.
+
+**mesos-dns** communicates with the leading Mesos master to build a DNS record set that reflects the tasks running in a Mesos cluster as documented here: http://mesosphere.github.io/mesos-dns/docs/naming.html.
+As of Kubernetes-Mesos [release v0.7.2](https://github.com/mesosphere/kubernetes/releases/tag/v0.7.2-v1.1.5) there is experimental support in the scheduler to populate a task's *discovery-info* field in order to generate alternative/more friendly record names in mesos-dns, for *pods* only.
+
+To enable this feature, set `--mesos-generate-task-discovery=true` when launching the scheduler.
```

Comment:
1. I expect most people to like this and am planning to make this the default in some future release (probably v0.8); for those people that don't like this -- now's their chance to complain/request some alternative
1. there's an impacting bug in `mesos-dns` https://github.com/mesosphere/mesos-dns/pull/382